app/code/Magento/InventoryInStorePickupQuote/Plugin/Checkout/AddPickupLocationAddressToShippingInformation.php

```diff
@@ -0,0 +1,144 @@
+use Magento\Store\Api\WebsiteRepositoryInterface;
+
+/**
+ * Assign Pickup Location Shipping Address for In-Store Pickup Quote.
+ */
+class AddPickupLocationAddressToShippingInformation
+{
+    /**
+     * @var GetPickupLocationInterface
+     */
+    private $getPickupLocation;
+
+    /**
+     * @var ToQuoteAddress
+     */
+    private $addressConverter;
+
+    /**
+     * @var CartRepositoryInterface
+     */
+    private $cartRepository;
+
+    /**
+     * @var StoreRepositoryInterface
+     */
+    private $storeRepository;
+
+    /**
+     * @var WebsiteRepositoryInterface
+     */
+    private $websiteRepository;
+
+    /**
+     * @var IsPickupLocationShippingAddress
+     */
+    private $isPickupLocationShippingAddress;
+
+    /**
+     * @var ExtractPickupLocationShippingAddressData
+     */
+    private $extractPickupLocationShippingAddressData;
+
+    /**
+     * @param GetPickupLocationInterface $getPickupLocation
+     * @param ToQuoteAddress $addressConverter
+     * @param CartRepositoryInterface $cartRepository
+     * @param StoreRepositoryInterface $storeRepository
+     * @param WebsiteRepositoryInterface $websiteRepository
+     * @param IsPickupLocationShippingAddress $isPickupLocationShippingAddress
+     * @param ExtractPickupLocationShippingAddressData $extractPickupLocationShippingAddressData
+     */
+    public function __construct(
+        GetPickupLocationInterface $getPickupLocation,
+        ToQuoteAddress $addressConverter,
+        CartRepositoryInterface $cartRepository,
+        StoreRepositoryInterface $storeRepository,
+        WebsiteRepositoryInterface $websiteRepository,
+        IsPickupLocationShippingAddress $isPickupLocationShippingAddress,
+        ExtractPickupLocationShippingAddressData $extractPickupLocationShippingAddressData
+    ) {
+        $this->getPickupLocation = $getPickupLocation;
+        $this->addressConverter = $addressConverter;
+        $this->cartRepository = $cartRepository;
+        $this->storeRepository = $storeRepository;
+        $this->websiteRepository = $websiteRepository;
+        $this->isPickupLocationShippingAddress = $isPickupLocationShippingAddress;
+        $this->extractPickupLocationShippingAddressData = $extractPickupLocationShippingAddressData;
+    }
+
+    /**
+     * Repalce Shipping Address with Pickup Location address for In-Store Pickup Delovery
+     *
+     * @param ShippingInformationManagementInterface $subject
+     * @param int $cartId
+     * @param ShippingInformationInterface $addressInformation
+     *
+     * @return array
+     * @throws NoSuchEntityException
+     * @throws StateException
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function beforeSaveAddressInformation(
+        ShippingInformationManagementInterface $subject,
+        int $cartId,
+        ShippingInformationInterface $addressInformation
+    ) {
+        $shippingCarrierCode = $addressInformation->getShippingCarrierCode();
```

Comment:
Very difficult to read the method.
Could you please refactor it or split to smaller parts to improve readability?

Reply:
...